### DISCUSSION TITLE
Tries to run tests in zig directories

### DISCUSSION THREAD

**Discussion by rockorager** [fredrikaverpil/neotest-golang#238] [General]
Hello! I'm trying out some neotest stuff via LazyVim. neotest-golang is working great for go files, but I also have a setup for zig - and neotest-golang tries to run tests when I am in a zig directory...I found the section below in the log, which made me suspicious that the golang adapter is being loaded in a directory which doesn't contain any go files. Is there a way to configure this / something wrong with the default LazyVim config for these adapters?

EDIT: I guess I should mention the problem :). When neotest-golang tries to run go tests in a zig directory, I get an error that there is no go.mod file - and this error seems to be preventing other test runners (neotest-zig) from running.

```
DEBUG | 2024-12-21T08:19:57Z-0600 | ...al/share/nvim/lazy/neotest/lua/neotest/adapters/init.lua:19 | Adapters: { {
    adapter = {
      _build_spec_with_buildfile = <function 1>,
      _build_spec_without_buildfile = <function 2>,
      _does_file_contain_tests = <function 3>,
      _get_neotest_result = <function 4>,
      _get_temp_file_path = <function 5>,
      _get_zig_symbol_name_from_node = <function 6>,
      _test_treesitter_query = "\t;;query\n    (test_declaration\n        [(identifier) (string)] @test.name\n    ) @test.definition\n",
      build_spec = <function 7>,
      dap = {
        adapter = ""
      },
      discover_positions = <function 8>,
      filter_dir = <function 9>,
      get_strategy_config = <function 10>,
      get_test_node_by_runspec = <function 11>,
      is_test_file = <function 12>,
      match_root_pattern = <function 13>,
      name = "neotest-zig",
      path_to_zig = "zig",
      results = <function 14>,
      root = <function 15>,
      setup = <function 16>,
      version = "v1.1.0",
      <metatable> = {
        __call = <function 17>
      }
    },
    root = "/home/tim/repos/libvaxis"
  }, {
    adapter = {
      build_spec = <function 18>,
      discover_positions = <function 19>,
      filter_dir = <function 20>,
      init = <function 21>,
      is_test_file = <function 22>,
      name = "neotest-golang",
      results = <function 23>,
      root = <function 24>,
      <metatable> = {
        __call = <function 25>
      }
    },
    root = "/home/tim/repos/libvaxis"
  } }
```

**Comment by fredrikaverpil** [comment]
Hey @rockorager and thanks for reaching out!

Would it be possible for you to provide a minimal reproducible example of this issue, as I'm not sure what I need to do to reproduce what you are seeing?

Like for example, maybe you can create a branch in which you add zig files into this very repo?

<img width="1862" alt="image" src="https://github.com/user-attachments/assets/64464f55-e18b-4096-b357-140adba2fe25" />

**Reply by fredrikaverpil** [reply]
@rockorager here's a branch `zig` where I've [added](https://github.com/fredrikaverpil/neotest-golang/commit/7b8734defa34a9313766c25072cedac561d66f0c) a root `build.zig` and then I added a `tests/go/main.zig`: https://github.com/fredrikaverpil/neotest-golang/tree/zig

I can't get go tests to trigger when I'm running tests from the `main.zig` file or when I'm running tests from the test summary pane. And running the already existing go tests works fine. So I can't really figure out what's going wrong on your end.

When you say you get an error about a missing `go.mod` when you run go tests, the intent here is that neotest-golang would search _upwards_ for a `go.mod`, starting from the `*_test.go` file containing the test. It searches up to `$HOME` but then it gives up.

**Reply by rockorager** [reply]
The error is intermittent for me...which makes it very difficult to debug :/

I am running neotest via LazyVim. I have the test.core, lang.go, and lang.zig installed. Just now I ran all test (<leader>tT) and got the error in [https://github.com/rockorager/vxfw-neovim](this repo) which has no go files.

I closed neovim, reopened it, and ran tests again. It worked just fine.

I didn't previously have any go files open :man_shrugging:

**Reply by fredrikaverpil** [reply]
I cloned down the repo and tried to repro this issue with `<leader>tT`. I also made sure both `zig` and `go` was on $PATH and available. But I can't see the `gopls` LSP kicking in and I can't see anything indicating the neotest-go adapter would be loaded.

I started Neovim/LazyVim in the repo root. I got errors when running `<leader>tT`, maybe because I'm on zig 0.13.0 and not the tip? 🤷 

Please file a bug if you can figure out a reproducible way to trigger something.

## LazyVim screenshots

<img width="2560" alt="Screenshot 2024-12-29 at 23 38 49" src="https://github.com/user-attachments/assets/baacf1e3-7279-4243-8afa-1d7407db19b9" />

<img width="2560" alt="Screenshot 2024-12-29 at 23 41 39" src="https://github.com/user-attachments/assets/725a77e7-6815-4e15-88c0-ba568a2cdc25" />

<img width="2560" alt="Screenshot 2024-12-29 at 23 43 24" src="https://github.com/user-attachments/assets/dd32baa2-a779-408d-a872-fecf01f05f59" />
